nav2_route/src/goal_intent_extractor.cpp

```diff
@@ -53,6 +58,49 @@ void GoalIntentExtractor::configure(
   nav2_util::declare_parameter_if_not_declared(
     node, "min_dist_from_start", rclcpp::ParameterValue(0.10));
   min_dist_from_start_ = static_cast<float>(node->get_parameter("min_dist_from_start").as_double());
+
+  nav2_util::declare_parameter_if_not_declared(
+    node, "num_of_nearest_nodes", rclcpp::ParameterValue(3));
+  int num_of_nearest_nodes = node->get_parameter("num_of_nearest_nodes").as_int();
+
+  node_spatial_tree_ = std::make_shared<NodeSpatialTree>(num_of_nearest_nodes);
+  node_spatial_tree_->computeTree(graph);
+
+  nav2_util::declare_parameter_if_not_declared(
+    node, "enable_search", rclcpp::ParameterValue(false));
+  enable_search_ = node->get_parameter("enable_search").as_bool();
+
+  if (enable_search_) {
+    std::string global_costmap_topic;
+    nav2_util::declare_parameter_if_not_declared(
+      node, "global_costmap_topic",
```

Comment:
Does this strictly need to be the global costmap?

Reply:
I think from a quick look it doesn't necessarily need to be

Reply:
Nope, 

but then the assumption would be that the local cost map always overlaps a node in the route graph.

Reply:
Sorry, I'm not saying we should change it to the local costmap frame, only that we should be able to accept it if someone wants to set it up that way. So the param shouldn't require the global costmap and handling for the local costmap should be considered